### PR TITLE
GraphQL: honor access control rules

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | todo

And now the GraphQL subsystem is hooked in the authorization mechanism.

Next step: validation, and we'll be almost done!

TODO:

* [x] Add some unit tests

### CODE REVIEW

File: features/authorization/deny.feature [hunk 1]
```diff
@@ -59,7 +58,7 @@ Feature: Authorization checking
     """
     Then the response status code should be 201
 
-  Scenario: An user retrieve cannot retrieve an item he doesn't own
+  Scenario: An user retrieves cannot retrieve an item he doesn't own
```
Comment (alanpoulain) [inline features/authorization/deny.feature]: Better? `An user cannot retrieve an item he doesn't own`

File: src/Graphql/Resolver/Factory/ItemMutationResolverFactory.php [hunk 2]
```diff
@@ -65,6 +68,12 @@ public function __invoke(string $resourceClass = null, string $rootClass = null,
             }
 
             $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
+            if (null !== $this->resourceAccessChecker) {
```
Comment (alanpoulain) [inline src/Graphql/Resolver/Factory/ItemMutationResolverFactory.php]: Shouldn't you put this in an abstract or a service? This is duplicated in `CollectionResolverFactory`.

Comment (alanpoulain) [reply src/Graphql/Resolver/Factory/ItemMutationResolverFactory.php]: And in `ItemResolver`.

Comment (dunglas) [reply src/Graphql/Resolver/Factory/ItemMutationResolverFactory.php]: I've moved this logic in a trait in the meantime.

File: tests/Security/EventListener/DenyAccessListenerTest.php [hunk 3]
```diff
@@ -63,16 +62,58 @@ public function testNoIsGrantedAttribute()
...
+        $resourceAccessCheckerProphecy->isGranted('Foo', 'has_role("ROLE_ADMIN")', Argument::type('array'))->willReturn(true)->shouldBeCalled();
+
+        $listener = $this->getListener($resourceMetadataFactoryProphecy->reveal(), $resourceAccessCheckerProphecy->reveal());
+        $listener->onKernelRequest($event);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Security\Core\Exception\AccessDeniedException
+     */
+    public function testIsNotGranted()
```
Comment (alanpoulain) [inline tests/Security/EventListener/DenyAccessListenerTest.php]: Use `@dataProvider` and remove the duplication?

Comment (alanpoulain) [reply tests/Security/EventListener/DenyAccessListenerTest.php]: You've done it below!

Comment (dunglas) [reply tests/Security/EventListener/DenyAccessListenerTest.php]: It should be done in the lowest branch or it will create merge conflicts later :(

**Pull request merged**